pixel/image.go

```diff
@@ -0,0 +1,223 @@
+package pixel
+
+import (
+	"unsafe"
+)
+
+// Image buffer, used for working with the native image format of various
+// displays. It works a lot like a slice: it can be rescaled while reusing the
+// underlying buffer and should be passed around by value.
+type Image[T Color] struct {
+	width  int16
+	height int16
+	data   unsafe.Pointer
+}
+
+// NewImage creates a new image of the given size.
+func NewImage[T Color](width, height int) Image[T] {
```

Comment:
Idea: might it be worth passing in the desired frame buffer to NewImage as a third argument? A couple of benefits I've come up with:
- Code readability improvements in generics department: Since the buffer has the color type there is no need to pass in a type parameter.
- Ease of use for people who don't want allocations: Makes it easier for users who want to reuse buffers and possible make images within images (given they use a new strided Image type for subimages)

If the user passes in a nil buffer then NewImage could automatically assign it a size. NewImage should however calculate the minimum size as it does now and make sure the argument buffer size meets the minimum size requirements.

Reply:
Interesting idea. Right now the buffer is an implementation detail, and I like to keep it that way (at least for now): calculating the size of the buffer is somewhat non-trivial for pixel sizes that don't neatly fit on byte boundaries (like RGB444 and perhaps others in the future like black&white with one bit per pixel). In fact RGB444 uses `[]byte` which wouldn't help with generics.

If there is a practical need for this (not just a theoretical one), we can always add a `NewImageWithBuffer` (name TBD) that can reuse an existing buffer.